storage/storage.go

```diff
@@ -0,0 +1,71 @@
+package storage
```

Comment:
If I told you to use a Database system instead of keeping all data in memory
1. Which kind of database would you choose and why?
2. Can you give a brief schema of this database (tables definition, relationships, or whatever you would do)
3. Would you use any ORM to connect and exchange data with the database? Do you have any in mind?

Reply:
Even thought I am most familiarized with relational databases and I would be more confident in implementing such a solution, I don’t think that this kind of db is suitable for the application. Instead  a no-SQL solution like MongoDB should be preferred.  Or maybe in a real world application an AWS-based  data structure could be used. Unfortunately I have no professional experience with no-SQL solutions but I will try to implement one.  
So taking these in mind an implementation for models is shown below:

```
package models

import (
    "go.mongodb.org/mongo-driver/bson/primitive"
)

type AssetType int

type Chart struct {
    Title string            `bson:"title" json:"title"`
    XAxis string            `bson:"x_axis" json:"x_axis"`
    YAxis string            `bson:"y_axis" json:"y_axis"`
    Data  map[string]float64 `bson:"data" json:"data"`
}

type Insight struct {
    Text string `bson:"text" json:"text"`
}

type Audience struct {
    Gender             string `bson:"gender" json:"gender"`
    BirthCountry       string `bson:"birth_country" json:"birth_country"`
    AgeGroup           string `bson:"age_group" json:"age_group"`
    HoursOnSocialMedia int    `bson:"hours_on_social_media" json:"hours_on_social_media"`
    PurchasesLastMonth int    `bson:"purchases_last_month" json:"purchases_last_month"`
}

type Asset struct {
    ID          primitive.ObjectID `bson:"_id,omitempty" json:"id,omitempty"`
    Type        AssetType          `bson:"type" json:"type"`
    Description string             `bson:"description" json:"description"`
    Chart       *Chart             `bson:"chart,omitempty" json:"chart,omitempty"`
    Insight     *Insight           `bson:"insight,omitempty" json:"insight,omitempty"`
    Audience    *Audience          `bson:"audience,omitempty" json:"audience,omitempty"`
}

type User struct {
    ID        primitive.ObjectID `bson:"_id,omitempty" json:"id,omitempty"`
    Favorites []Asset            `bson:"favorites" json:"favorites"`
}

```
After the models are defined indexes should be created to improve performance. For example indexes could be created on the favorites type field and the favorites id field.  Also we could implement some validation logic making sure that values like User.Id and Asset.Id are not empty before performing any operation. 
In the application I have used the echo framework so as for the use of an ORM to interact with a MongoDB, the MongoDB Go Driver could be a suitable solution providing all the necessary features to manage complex data structures